packages/react-dom/src/__tests__/ReactDOMEventListener-test.js

```diff
@@ -271,46 +292,93 @@ describe('ReactDOMEventListener', () => {
     document.body.removeChild(container);
   });
 
+  // This is a special case for submit and reset events as they are listened on
+  // at the element level and not the document.
+  // @see https://github.com/facebook/react/pull/13462
+  it('should not receive submit events if native, interim DOM handler prevents it', () => {
```

Comment:
This test indeed fails if I re-apply my changes from #13358. If the `onSubmit` handler added a `preventDefault()`, this would never be called.